redis/sentinel.py

```diff
@@ -97,14 +94,12 @@ def reset(self):
         self.slave_rr_counter = None
 
     def get_master_address(self):
+        """Get the address of the current master"""
         master_address = self.sentinel_manager.discover_master(
             self.service_name)
         if self.is_master:
-            if self.master_address is None:
+            if master_address != self.master_address:
                 self.master_address = master_address
-            elif master_address != self.master_address:
-                # Master address changed, disconnect all clients in this pool
-                self.disconnect()
```

Comment:
All old connections will get closed when attempting to fetch them from the pool. I could theoretically close connections from `_available_connections` here, but I don't know what would the thread-safety implications would be. The fact that this code depends on atomic operations in Python is confusing enough.